src/server/nuclearnet/web_socket_proxy_nuclearnet_server.ts

```diff
@@ -93,6 +102,83 @@ class WebSocketServerClient {
   }
 
   private onPacket = (event: string, packet: NUClearNetPacket) => {
+    const peerKey = this.getPeer(this.processors.keys(), packet.peer)
+    if (peerKey) {
+      const processor = this.processors.get(peerKey)
+      if (processor) {
+        processor.onPacket(event, packet)
+      }
+    }
+  }
+
+  /**
+   * Look for the needle 'peer' inside the haystack 'peers'. They might not be the same object reference.
```

Comment:
how could you! 😭 😛 

haha well, _will not_ in the current use case, but this method could take any list of peers and any peer so I wanted to remain ambiguous.